### DISCUSSION TITLE
Solidus: Stock Managment

### DISCUSSION THREAD

**Discussion by shakya-rasna** [solidusio/solidus#4524] [Troubleshooting] [answered]
II am facing issue on solidus checkout flow. i have removed solidus delivery checkout step and now my stock are not being updated after checkout completed and payment state is 'paid'. What am i missing, how can we update inventory?

**Comment by jarednorman** [comment, marked as answer]
Inventory isn't deducted until the inventory units for the shipments get finalized. Removing the delivery step is usually done for stores selling digital goods that don't track quantity. Why have you removed it, given (I'm assuming) that you're selling physical goods?

**Reply by shakya-rasna** [reply]
Yes we have local pick up so we are not handling shipments. For this case how can we handle stock management

**Reply by jarednorman** [reply]
If you don't create shipments/inventory units, inventory will never get deducted. You'll need to either switch to an approach that uses shipments or you'll need to add an additional step to your state machine that deducts the inventory.

**Reply by shakya-rasna** [reply]
Can u guide us the way to deduct the inventory without shipment?

**Reply by jarednorman** [reply]
I would consider hooking into the order state machine and deduct the inventory on transition to complete.
